ghcjs-src/Miso/FFI.hs

```diff
@@ -101,23 +104,41 @@ foreign import javascript unsafe "console.log($1);"
 foreign import javascript unsafe "$r = JSON.stringify($1);"
   stringify' :: JSVal -> IO JSString
 
-foreign import javascript unsafe "$r = JSON.parse($1);"
+-- | Parse a raw string into JSON. Accepts any JS value.
+foreign import javascript unsafe "$r = JSON.parse('' + $1);"
   parse' :: JSVal -> IO JSVal
 
 -- | Converts a JS object into a JSON string
 stringify :: ToJSON json => json -> IO JSString
 {-# INLINE stringify #-}
 stringify j = stringify' =<< toJSVal (toJSON j)
 
--- | Parses a JSString
+-- | Parses a JSVal (which of course should be a string). Less safe
+-- | than 'parseString'.
+parseString :: FromJSON json => JSString -> IO json
+{-# INLINE parseString #-}
+parseString = parse . unsafeCoerce
```

Comment:
No need for `unsafeCoerce`, you can use `jsval` here.

Reply:
Didn't know about that one, thanks!